app/api/route/models/param_schema.py

```diff
@@ -0,0 +1,111 @@
+import inspect
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Iterable, Optional, Type
+
+from apiflask.validators import OneOf, Regexp, Validator
+
+import api.logging
+
+logger = api.logging.get_logger(__name__)
+
+PARAM_UNSET_IN_REQUEST: str = "PARAM_UNSET_IN_REQUEST"
+
+
+@dataclass
+class ParamFieldConfig:
+    example: Optional[str | bool | int] = None
+    description: Optional[str] = None
+    allowed_values: Optional[Iterable | Type[Enum]] = None
+    regex: Optional[str] = None
+
+    def build(self, **overrides: Any) -> Any:
```

Comment:
This whole ParamFieldConfig class is attempting to make building the configuration for examples/descriptions/regexes/etc. a lot simpler, but let me know if you have any suggestions to organize this differently.

For example, lets say you want to validate a phone number has the right format, and provide a default. If you do that directly in your model, you'd need to do:

```py
...
    phone_number = field(metadata={
        "metadata":{"example":"123-456-7890", "description":"it's a phone number!"},
        "validate": [Regexp("... your regex")]
    }
```
Which is a little oddly formatted, as you've got two metadata objects which contain different info. To make it simpler, this util makes it so you can just pass in simple params which will auto-convert to that.

Additionally some fields probably want the exact same config approach (`created_at` and `updated_at` are both just datetime), so we can share that config, which is what the ParamSchema class below is doing. The build function itself takes in overrides for some fields in case you want to use say `first_name` and make the description more specific.

Note that we cannot do:
```py
class ParamSchema:
     datetime = field(...)
```
as if you have multiple fields reference the same call to the field function, it just isn't happy and only displays one in Swagger (guessing some sort of binding underneath the hood?).